### PR TITLE
Create CI Lane for Mono in Interpreter Mode running the runtime tests

### PR DESCRIPTION
**Pull request by naricc**
Create CI Lane for Mono in Interpreter mode running the runtime tests. To do this we need to ensure that the environment variable MONO_ENV_OPTIONS contains "--interpreter" when running the tests.

### CODE REVIEW

File: eng/pipelines/common/templates/runtimes/run-test-job.yml [hunk 1]
```diff
@@ -75,12 +76,12 @@ jobs:
...
+      displayName: '${{ parameters.runtimeFlavorDisplayName }} ${{ parameters.runtimeModeDisplayName}} Pri0 Runtime Tests Run ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
 
     ${{ if ne(parameters.testGroup, 'innerloop') }}:
       name: 'run_test_p1_${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'
-      displayName: '${{ parameters.runtimeFlavorDisplayName }} Pri1 Test Run ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
+      displayName: '${{ parameters.runtimeFlavorDisplayName }} ${{ parameters.runtimeModeDisplayName }} Pri1 Runtime Tests Run ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
```
Comment (safern) [inline eng/pipelines/common/templates/runtimes/run-test-job.yml:84]: runtimeModeDisplayName is not defined as a parameter of the template.

File: eng/pipelines/common/templates/runtimes/send-to-helix-step.yml [hunk 2]
```diff
@@ -57,6 +58,7 @@ steps:
       _TimeoutPerTestCollectionInMinutes: ${{ parameters.timeoutPerTestCollectionInMinutes }}
       _TimeoutPerTestInMinutes: ${{ parameters.timeoutPerTestInMinutes }}
       runtimeFlavorDisplayName: ${{ parameters.runtimeFlavorDisplayName }}
+      _RuntimeMode: #{{ parameters.runtimeMode }}
```
Comment (safern) [inline eng/pipelines/common/templates/runtimes/send-to-helix-step.yml:61]: Is this only relevant to windows?

Comment (safern) [reply eng/pipelines/common/templates/runtimes/send-to-helix-step.yml:61]: Also, you have a typo, `#`->`$`.

File: src/coreclr/tests/helixpublishwitharcade.proj [hunk 3]
```diff
@@ -223,11 +224,19 @@
     <HelixPreCommand Include="export __CrashDumpFolder=$HELIX_DUMP_FOLDER" Condition=" '$(TargetOS)' != 'OSX' " />
     <HelixPreCommand Include="export __CrashDumpFolder=/cores" Condition=" '$(TargetOS)' == 'OSX' " /> <!-- Helix doesn't specify the dump folder for OSX 10.14, so we need to manually specify it. Tracked by dotnet/core-eng#7872 -->
     <HelixPreCommand Include="cat $__TestEnv" />
-
     <HelixPreCommand Include="sudo bash -c 'echo $HELIX_DUMP_FOLDER/core.%u.%p > /proc/sys/kernel/core_pattern'" Condition=" '$(TargetOS)' != 'OSX' " />
     <HelixPreCommand Include="ulimit -c unlimited" Condition=" '$(TargetOS)' == 'OSX' " />
   </ItemGroup>
 
+  <ItemGroup Condition=" '$(TargetsWindows)' != 'true' and $(RuntimeMode) == 'interpreter' " >
```
Comment (safern) [inline src/coreclr/tests/helixpublishwitharcade.proj:231]: You could just put these to in the `ItemGroup` above by conditioning the individual Item inclusion as we do in other commands.

```xml
<HelixPreCommand Include="...." Condition="'$(RuntimeMode)' == 'interpreter'" />
```

Also, `$(RuntimeMode)` needs to be enclosed in `'` in the condition.

File: src/coreclr/tests/helixpublishwitharcade.proj [hunk 4]
```diff
@@ -206,6 +207,7 @@
     <HelixPreCommand Include="set __CollectDumps=1" />
     <HelixPreCommand Include="set __CrashDumpFolder=%HELIX_DUMP_FOLDER%" />
     <HelixPreCommand Include="type %__TestEnv%" />
+    <HelixPreCommand Include="set MONO_ENV_OPTIONS='--interpreter' " Condition=" '$(RuntimeMode)' == 'interpreter' " />
```
Comment (safern) [inline src/coreclr/tests/helixpublishwitharcade.proj:210]: Nit: remove space before `"`.

```suggestion
    <HelixPreCommand Include="set MONO_ENV_OPTIONS='--interpreter'" Condition=" '$(RuntimeMode)' == 'interpreter'" />
```

File: src/coreclr/tests/helixpublishwitharcade.proj [hunk 5]
```diff
@@ -223,9 +225,9 @@
     <HelixPreCommand Include="export __CrashDumpFolder=$HELIX_DUMP_FOLDER" Condition=" '$(TargetOS)' != 'OSX' " />
     <HelixPreCommand Include="export __CrashDumpFolder=/cores" Condition=" '$(TargetOS)' == 'OSX' " /> <!-- Helix doesn't specify the dump folder for OSX 10.14, so we need to manually specify it. Tracked by dotnet/core-eng#7872 -->
     <HelixPreCommand Include="cat $__TestEnv" />
-
     <HelixPreCommand Include="sudo bash -c 'echo $HELIX_DUMP_FOLDER/core.%u.%p > /proc/sys/kernel/core_pattern'" Condition=" '$(TargetOS)' != 'OSX' " />
     <HelixPreCommand Include="ulimit -c unlimited" Condition=" '$(TargetOS)' == 'OSX' " />
+    <HelixPreCommand Include="export MONO_ENV_OPTIONS='--interpreter" Condition=" '$(RuntimeMode)' == 'interpreter' " />
```
Comment (safern) [inline src/coreclr/tests/helixpublishwitharcade.proj:230]: Nit:
```suggestion
    <HelixPreCommand Include="export MONO_ENV_OPTIONS='--interpreter" Condition=" '$(RuntimeMode)' == 'interpreter'" />
```

File: eng/pipelines/common/templates/runtimes/run-test-job.yml [hunk 6]
```diff
@@ -75,12 +76,12 @@ jobs:
 
     # Compute job name from template parameters
     ${{ if eq(parameters.testGroup, 'innerloop') }}:
-      name: 'run_test_p0_${{ parameters.runtimeFlavor }}_${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'
-      displayName: '${{ parameters.runtimeFlavorDisplayName }}  Pri0 Test Run ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
+      name: 'run_test_p0_${{ parameters.runtimeFlavor }}${{ parameters.runtimeMode }}_${{ parameters.displayNameArgs }}_${{ parameters.osGroup }}${{ parameters.osSubgroup }}_${{ parameters.archType }}_${{ parameters.buildConfig }}'
+      displayName: '${{ parameters.runtimeFlavorDisplayName }} ${{ parameters.runtimeModeDisplayName}} Pri0 Runtime Tests Run ${{ parameters.displayNameArgs }} ${{ parameters.osGroup }}${{ parameters.osSubgroup }} ${{ parameters.archType }} ${{ parameters.buildConfig }}'
```
Comment (safern) [inline eng/pipelines/common/templates/runtimes/run-test-job.yml:80]: I don't understand where you're getting `runtimeModeDisplayName` from. I think you have to do something similar to what we do for `runtimeFlavorDisplayName`

**Comment by SamMonoRT** [comment]
LGTM - @safern any concerns before we can merge this ?

**Comment by safern** [comment]
I’ll take a look in a few mins.

File: eng/pipelines/runtime.yml [hunk 7]
```diff
@@ -716,6 +716,29 @@ jobs:
...
+    platforms:
+    - OSX_x64
+    helixQueueGroup: pr
+    helixQueuesTemplate: /eng/pipelines/coreclr/templates/helix-queues-setup.yml
+    jobParameters:
+      testGroup: innerloop
+      liveLibrariesBuildConfig: ${{ variables.debugOnPrReleaseOnRolling }}
+      liveRuntimeBuildConfig: release
+      runtimeMode: 'interpreter'
+      runtimeModeDisplayName: "Interpreter"
```
Comment (safern) [inline eng/pipelines/runtime.yml:736]: Why use double quotes here?

File: src/coreclr/tests/helixpublishwitharcade.proj [hunk 8]
```diff
@@ -206,6 +207,7 @@
     <HelixPreCommand Include="set __CollectDumps=1" />
     <HelixPreCommand Include="set __CrashDumpFolder=%HELIX_DUMP_FOLDER%" />
     <HelixPreCommand Include="type %__TestEnv%" />
+    <HelixPreCommand Include="set MONO_ENV_OPTIONS='--interpreter'" Condition=" '$(RuntimeMode)' == 'interpreter' " />
```
Comment (safern) [inline src/coreclr/tests/helixpublishwitharcade.proj:210]: I think the TestRunNamePrefix should also contain `interpreter` in the name to make it easier to identify if there is a failure in the azure devops test explorer.

**Review by safern — APPROVED** [review]
Other than my two comments LGTM

**Review by safern — APPROVED** [review]
